src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -172,77 +212,158 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
+            {
+                hasErrors = true;
+                resultKind = LookupResultKind.NotAValue;
+            }
+
+            return new BoundPropertyPatternMember(
+                memberName,
+                boundMember.ExpressionSymbol,
+                arguments,
+                argumentNamesOpt,
+                argumentRefKindsOpt,
+                expanded,
+                argsToParamsOpt,
+                resultKind,
+                boundMember.Type,
+                hasErrors);
         }
 
-        private Symbol FindPropertyOrFieldByName(TypeSymbol type, SyntaxToken name, out LookupResultKind resultKind, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        private BoundExpression BadSubpatternMemberAccess(
+            BoundExpression boundMember,
+            BoundImplicitReceiver implicitReceiver,
+            IdentifierNameSyntax memberName,
+            DiagnosticBag diagnostics,
+            bool suppressErrors)
         {
-            var symbols = ArrayBuilder<Symbol>.GetInstance();
-            var lookupResult = LookupResult.GetInstance();
-            this.LookupMembersWithFallback(lookupResult, type, name.ValueText, arity: 0, useSiteDiagnostics: ref useSiteDiagnostics);
-            resultKind = lookupResult.Kind;
-            Symbol result = null;
-
-            if (lookupResult.IsMultiViable)
+            if (!suppressErrors)
             {
-                foreach (var symbol in lookupResult.Symbols)
+                string member = memberName.Identifier.ValueText;
+                switch (boundMember.ResultKind)
                 {
-                    if (symbol.Kind == SymbolKind.Property || symbol.Kind == SymbolKind.Field)
-                    {
-                        if (result != null && symbol != result)
-                        {
-                            resultKind = LookupResultKind.Ambiguous;
-                            result = null;
-                            break;
-                        }
-                        else
-                        {
-                            result = symbol;
-                        }
-                    }
+                    case LookupResultKind.Empty:
+                        Error(diagnostics, ErrorCode.ERR_NoSuchMember, memberName, implicitReceiver.Type, member);
+                        break;
+
+                    case LookupResultKind.Inaccessible:
+                        boundMember = CheckValue(boundMember, BindValueKind.RValue, diagnostics);
+                        Debug.Assert(boundMember.HasAnyErrors);
+                        break;
+
+                    default:
+                        Error(diagnostics, ErrorCode.ERR_PropertyLacksGet, memberName, member);
+                        break;
                 }
             }
 
-            lookupResult.Free();
-            return result;
+            return ToBadExpression(boundMember, LookupResultKind.Inaccessible);
```

Comment:
There is an open LDM issue for how this lookup should work. When we have more guidance we can complete all the specified scenarios.